aibolit/patterns/supermethod/supermethod.py

```diff
@@ -1,15 +1,30 @@
-from javalang.parse import parse
-from javalang.tree import MethodDeclaration, SuperMethodInvocation
-
+from aibolit.utils.ast import AST, ASTNodeType
+from aibolit.utils.java_package import JavaPackage
 from aibolit.utils.encoding_detector import read_text_with_autodetected_encoding
+import re
+from typing import Set
 
 
 class SuperMethod:
 
     def __init__(self):
         pass
 
-    def value(self, filename: str):
+    def _get_node_name(self, ast: AST, node: int) -> str:
+        extracted_name = None
+        names = ast.children_with_type(node, ASTNodeType.STRING)
+        for each_string in names:
+            method_name = ast.get_attr(each_string, 'string')
+            # Checking not to start with '/' is aimed to get
+            # rid of comments, which are all childs of node.
+            # We check the occurance any letter in name in order
+            # to get rid of '' string and None.
+            if not method_name.startswith('/') and re.search(r'[^\W\d]', method_name) is not None:
+                extracted_name = method_name
+                return extracted_name
+        return ''
+
+    def value(self, filename: str) -> Set[int]:
```

Comment:
Don't we usually return lists in patterns?

Reply:
yeap, agree